### PR TITLE
refactor: move helper functions from pages

### PR DESCRIPTION
**Pull request by nicolearagao**
Move utility functions to helpers module, improve error handling, add tests and docstrings.

### CODE REVIEW

**Review by ruda — APPROVED** [review]
Looks good.

File: src/common/helpers.ts [hunk 1]
```diff
@@ -265,6 +266,56 @@ const isIpAddress = (name: string): boolean => {
   return false;
 };
 
+const getLastUpdated = (credential: CredentialType): string => {
```
Comment (ruda) [inline src/common/helpers.ts:269]: Is there an unit test for this? If not, do you think worth it have one?

File: src/common/helpers.ts [hunk 2]
```diff
@@ -265,6 +266,56 @@ const isIpAddress = (name: string): boolean => {
...
+  } else if (days < 30) {
+    return `${days} days ago`;
+  } else if (months === 1) {
+    return 'A month ago';
+  } else if (months < 12) {
+    return `${months} months ago`;
+  } else if (years === 1) {
+    return 'A year ago';
+  } else {
+    return `${years} years ago`;
```
[the remark this answers is not in the document]
Comment (infinitewarp) [reply src/common/helpers.ts:301]: (I understand you're just moving code around, but the question still stands.) 🙂

[the remark this answers is not in the document]
Comment (bruno-fs) [reply src/common/helpers.ts:301]: old UI did that. (not saying the new should)

File: src/common/helpers.ts (hunk 1 above)
Comment (nicolearagao) [reply src/common/helpers.ts:269]: yes, I def want to cover these in tests.  I am just going step by step. =)

File: src/common/helpers.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gitdallas) [reply src/common/helpers.ts:301]: don't really need the `else`s here as the `return`s will stop execution... again not your fault, code just moved :P

[the remark this answers is not in the document]
Comment (gitdallas) [reply src/common/helpers.ts:301]: also, i notice that `SourceListView` has a `getTimeDisplayHowLongAgo` function. wonder if these could be combined or if this format is different and needs to be?

**Review by gitdallas — APPROVED** [review]

**Review by infinitewarp — APPROVED** [review]

**Pull request merged**